packages/rollup-contracts/contracts/ovm/PartialStateManager.sol

```diff
@@ -0,0 +1,329 @@
+     * then deploy the execution manager (passing in the state manager address), and then set the execution manager
+     * address in the state manager. This is a bit ugly & probably should be thought through a bit more.
+     */
+    function setExecutionManager(address _executionManagerAddress) public {
+        require(msg.sender == address(fraudVerifier));
+        executionManager = ExecutionManager(_executionManagerAddress);
+    }
+
+    /**
+     * @notice Initialize a new transaction execution
+     */
+    function initNewTransactionExecution() external {
+        require(msg.sender == address(fraudVerifier));
+        existsInvalidStateAccess = false;
+        updatedStorageCounter = 0;
+        updatedContractsCounter = 0;
+    }
+
+    function setIsTransitionSuccessfullyExecuted(bool _isTransactionSuccessfullyExecuted) external {
+        require(msg.sender == address(fraudVerifier));
+        isTransactionSuccessfullyExecuted = _isTransactionSuccessfullyExecuted;
+    }
+
+    function ensureVerifiedStorage(address _ovmContractAddress, bytes32 _slot) private {
+        if (!isVerifiedStorage[_ovmContractAddress][_slot]) {
+            existsInvalidStateAccess = true;
+        }
+    }
+
+    function ensureVerifiedContract(address _ovmContractAddress) private {
+        if (!isVerifiedContract[_ovmContractAddress]) {
+            existsInvalidStateAccess = true;
+        }
+    }
+
+    /****************
+    * Pre-Execution *
+    ****************/
+
+    function verifyStorage(address _ovmContractAddress, bytes32 _slot, bytes32 _value) external preExecution {
+        // TODO: Verify inclusion proof against root
+
+        isVerifiedStorage[_ovmContractAddress][_slot] = true;
+        ovmContractStorage[_ovmContractAddress][_slot] = _value;
+    }
+
+    function verifyContract(address _ovmContractAddress, bytes32 _codeHash, uint _nonce) external preExecution {
+        // TODO: Verify inclusion proof against root
+        // TODO: Get the actual code contract addr from the registry
+        address _codeContractAddress = 0x0000000000000000000000000000000000000000;
+
+        isVerifiedContract[_ovmContractAddress] = true;
+        ovmContractNonces[_ovmContractAddress] = _nonce;
+        associateCodeContract(_ovmContractAddress, _codeContractAddress);
+    }
+
+    /*****************
+    * Post-Execution *
+    *****************/
+
+    // TODO:
+    function updateStorageNode(address _ovmContractAddress, bytes32 _slot, bytes32 _value) external postExecution {
+        // TODO: Finish filling out this function...
+        // Get the next storage we need to update using the updatedStorageCounter
+        address contractAddressToUpdate = address(bytes20(updatedStorage[updatedStorageCounter - 1]));
+        bytes32 storageKeyToUpdate = updatedStorage[updatedStorageCounter];
+        bytes32 storageValueToUpdate = getStorage(contractAddressToUpdate, storageKeyToUpdate);
```

Comment:
make two mappings

Reply:
also avoid double updates of the same key

Reply:
Turns out we don't actually have to do this! because we're always getting the most recent value with getStorage